pkg/sdk/grants_impl.go

```diff
@@ -266,16 +266,21 @@ func (v *grants) Show(ctx context.Context, opts *ShowGrantOptions) ([]Grant, err
 		// SHOW GRANTS of DATABASE ROLE requires a special handling:
 		// - it returns no account name, so for other SHOW GRANTS types it needs to be skipped
 		// - it returns fully qualified name for database objects
-		if !(valueSet(opts.Of) && valueSet(opts.Of.DatabaseRole)) {
+		granteeNameRaw := dbRows[i].GranteeName
+		if !(valueSet(opts.Of) && valueSet(opts.Of.DatabaseRole)) { //nolint:gocritic
+			granteeName := granteeNameRaw
 			if grant.GrantedTo == ObjectTypeShare {
-				oldId := grant.GranteeName.Name()
-				skipAccount := oldId[strings.IndexRune(oldId, '.')+1:]
-				resultList[i].GranteeName = NewAccountObjectIdentifier(skipAccount)
+				granteeName = granteeName[strings.IndexRune(granteeName, '.')+1:]
 			}
-		} else {
-			if grant.GrantedTo != ObjectTypeRole && grant.GrantedTo != ObjectTypeShare {
-				resultList[i].GranteeName = NewDatabaseObjectIdentifierFromFullyQualifiedName(grant.GranteeName.FullyQualifiedName())
+			resultList[i].GranteeName = NewAccountObjectIdentifier(granteeName)
+		} else if !slices.Contains([]ObjectType{ObjectTypeRole, ObjectTypeShare}, grant.GrantedTo) {
+			id, err := ParseDatabaseObjectIdentifier(granteeNameRaw)
+			if err != nil {
+				return nil, err
 			}
+			resultList[i].GranteeName = id
+		} else {
+			resultList[i].GranteeName = NewAccountObjectIdentifier(granteeNameRaw)
```

Comment:
The database role case is fixed, but I think the problem still remains for the other cases, now I'm thinking that maybe a not that bad solution would be to just pass raw grantee name instead of packing it with NewAccountObjectIdentifier? I mean, `&sdk.AccountObjectIdentifier { granteeNameRaw }` (?)

Reply:
As discussed, it should behave correctly